syntax = "proto3";

package tvsc.discovery;

message ServiceDescriptor {
  // This service_type definition allows for any service type. Please use service type helper
  // methods/static definitions/etc. to constrain the names of the services to the set of types that
  // are interesting to the application.
  string service_type = 1;
  // Which interfaces should this service listen on. These are currently interface names, but per
  // https://agnat.github.io/node_mdns/user_guide.html#network_interfaces, this approach may not
  // work on all platforms.
  repeated string allowed_interface_names = 2;
  int32 port = 3;
}

message ServiceSet {
  string canonical_name = 1;
  string published_name = 2;
  repeated ServiceDescriptor services = 3;
}
